CPAC/nuisance/nuisance.py

```diff
@@ -496,6 +496,115 @@ def gather_nuisance(
     return output_file_path, censor_indices
 
 
+def offending_timepoints_connector(
+    nuisance_selectors, name="offending_timepoints_connector"
+):
+    inputspec = pe.Node(
+        util.IdentityInterface(
+            fields=[
+                "fd_j_file_path",
+                "fd_p_file_path",
+                "dvars_file_path",
+            ]
+        ),
+        name="inputspec",
+    )
+
+    wf = pe.Workflow(name=name)
+
+    outputspec = pe.Node(
+        util.IdentityInterface(fields=["out_file"]),
+        name="outputspec",
+    )
+
+    censor_methods = ["Kill", "Zero", "Interpolate", "SpikeRegression"]
+
+    censor_selector = nuisance_selectors.get("Censor")
+    if censor_selector is None or censor_selector.get("method") not in censor_methods:
+        msg = (
+            "Improper censoring method specified ({0}), "
+            "should be one of {1}.".format(
+                censor_selector.get("method", None) if censor_selector else None,
+                censor_methods,
+            )
+        )
+        raise ValueError(msg)
```

Comment:
Let's move this validation to the schema! We can drop this all the way down to
```suggestion
    censor_selector = nuisance_selectors.get("Censor")
```
and change https://github.com/FCP-INDI/C-PAC/blob/0cc9264ae521db40e58aded42c5336417f1409a6/CPAC/pipeline/schema.py#L1039 to 
```Python
                                    "method": In(["Kill", "Zero", "Interpolate", "SpikeRegression"]),
```

to catch this right at the beginning of the run

Reply:
Yesss great idea.